File: server/lib/cartography/field_card.ex
defmodule Cartography.FieldCard do
  use Ecto.Schema

  @primary_key false
  @derive Jason.Encoder

  schema "field_cards" do
    field(:card_id, :integer, primary_key: true)
    field(:account_id, :string)
    field(:field_id, :integer)
    field(:grid_x, :integer)
    field(:grid_y, :integer)
  end
end

File: src/lib/appserver/FieldCard.ts
import type { AccountId } from "./Account";
import type { CardId } from "./Card";
import type { FieldId } from "./Field";

export interface FieldCard {
  // eslint-disable-next-line @typescript-eslint/naming-convention -- this is a server owned field
  account_id: AccountId;
  // eslint-disable-next-line @typescript-eslint/naming-convention -- this is a server owned field
  card_id: CardId;
  // eslint-disable-next-line @typescript-eslint/naming-convention -- this is a server owned field
  field_id?: FieldId;
  // eslint-disable-next-line @typescript-eslint/naming-convention -- this is a server owned field
  grid_x?: number;
  // eslint-disable-next-line @typescript-eslint/naming-convention -- this is a server owned field
  grid_y?: number;
}
